### ISSUE TITLE
Coq 8.10

### ISSUE THREAD

**Issue by silene** [coq/coq#8446]
## As soon as the previous version branched off master ##

- [x] Create a new issue to track the release process where you can copy-paste
  the present checklist.
- [x] Change the version name to the next major version and the magic
  numbers (see [#7008](https://github.com/coq/coq/pull/7008/files)).

  Additionally, in the same commit, update the compatibility
  infrastructure, which consists of invoking
  [`dev/tools/update-compat.py`](../tools/update-compat.py) with the
  `--master` flag.

  Note that the `update-compat.py` script must be run twice: once
  *immediately after* branching with the `--master` flag (which sets
  up Coq to support four `-compat` flag arguments), *in the same
  commit* as the one that updates `coq_version` in
  [`configure.ml`](../../configure.ml), and once again later on before
  the next branch point with the `--release` flag (see next section).
- [x] Put the corresponding alpha tag using `git tag -s`.
  The `VX.X+alpha` tag marks the first commit to be in `master` and not in the
  branch of the previous version.
- [x] Create the `X.X+beta1` milestone if it did not already exist.
- [x] Decide the release calendar with the team (freeze date, beta date, final
  release date) and put this information in the milestone (using the
  description and due date fields).

## Anytime after the previous version is branched off master ##

- [ ] Update the compatibility infrastructure to the next release,
  which consists of invoking
  [`dev/tools/update-compat.py`](../tools/update-compat.py) with the
  `--release` flag; this sets up Coq to support three `-compat` flag
  arguments.  To ensure that CI passes, you will have to decide what
  to do about all test-suite files which mention `-compat U.U` or
  `Coq.Comapt.CoqUU` (which is no longer valid, since we only keep
  compatibility against the two previous versions on releases), and
  you may have to prepare overlays for projects using the
  compatibility flags.

## About one month before the beta ##

- [ ] Create the `X.X.0` milestone and set its due date.
- [ ] Send an announcement of the upcoming freeze on Coqdev and ask people to
  remove from the beta milestone what they already know won't be ready on time
  (possibly postponing to the `X.X.0` milestone for minor bug fixes,
  infrastructure and documentation updates).
- [ ] Determine which issues should / must be fixed before the beta, add them
  to the beta milestone, possibly with a
  ["priority: blocker"](https://github.com/coq/coq/labels/priority%3A%20blocker)
  label. Make sure that all these issues are assigned (and that the assignee
  provides an ETA).
- [ ] Ping the development coordinator (**@mattam82**) to get him started on
  the update to the Credits chapter of the reference manual.
  See also [#7058](https://github.com/coq/coq/issues/7058).
  The command to get the list of contributors for this version is
  `git shortlog -s -n VX.X+alpha..master | cut -f2 | sort -k 2`
  (the ordering is approximative as it will misplace people with middle names).

## On the date of the feature freeze ##

- [ ] Create the new version branch `vX.X` (using this name will ensure that
  the branch will be automatically protected).
- [ ] Remove all remaining unmerged feature PRs from the beta milestone.
- [ ] Start a new project to track PR backporting. The proposed model is to
  have a "X.X-only PRs" column for the rare PRs on the stable branch, a
  "Request X.X inclusion" column for the PRs that were merged in `master` that
  are to be considered for backporting, a "Waiting for CI" column to put the
  PRs in the process of being backported, and "Shipped in ..." columns to put
  what was backported. (The release manager is the person responsible for
  merging PRs that target the version branch and backporting appropriate PRs
  that are merged into `master`).
  A message to **@coqbot** in the milestone description tells it to
  automatically add merged PRs to the "Request X.X inclusion" column.
- [ ] Delay non-blocking issues to the appropriate milestone and ensure
  blocking issues are solved. If required to solve some blocking issues,
  it is possible to revert some feature PRs in the version branch only.

## Before the beta release date ##

- [ ] Ensure the Credits chapter has been updated.
- [ ] Ensure that an appropriate version of the plugins we will distribute with
  Coq has been tagged.
- [ ] Have some people test the recently auto-generated Windows and MacOS
  packages.
- [ ] Change the version name from alpha to beta1 (see
  [#7009](https://github.com/coq/coq/pull/7009/files)).
  We generally do not update the magic numbers at this point.
- [ ] Put the `VX.X+beta1` tag using `git tag -s`.

### These steps are the same for all releases (beta, final, patch-level) ###

- [ ] Send an e-mail on Coqdev announcing that the tag has been put so that
  package managers can start preparing package updates (including a
  `coq-bignums` compatible version).
- [ ] Ping **@erikmd** to update the Docker images in `coqorg/coq`
  (requires `coq-bignums` in `extra-dev` for a beta / in `released`
  for a final release).
- [ ] Draft a release on GitHub.
- [ ] Get **@maximedenes** to sign the Windows and MacOS packages and
  upload them on GitHub.
- [ ] Prepare a page of news on the website with the link to the GitHub release
  (see [coq/www#63](https://github.com/coq/www/pull/63)).
- [ ] Upload the new version of the reference manual to the website.
  *TODO: setup some continuous deployment for this.*
- [ ] Merge the website update, publish the release
  and send annoucement e-mails.
- [ ] Ping **@Zimmi48** to publish a new version on Zenodo.
  *TODO: automate this.*
- [ ] Close the milestone

## At the final release time ##

- [ ] Change the version name to X.X.0 and the magic numbers (see
  [#7271](https://github.com/coq/coq/pull/7271/files)).
- [ ] Put the `VX.X.0` tag.

Repeat the generic process documented above for all releases.

- [ ] Switch the default version of the reference manual on the website.

**Comment by JasonGross** [comment]
Note that this checklist will change when #8374 is merged

**Comment by Zimmi48** [comment]
#8374 has been replaced by #8637 which has been merged. So the beginning of this checklist should be updated to reflect that.

**Comment by JasonGross** [comment]
I have updated the checklist with the version currently on master.  I have not changed which things are checked off, though I am suspicious of the current checking.

**Comment by JasonGross** [comment]
Is the version number on master going to be bumped to 8.11+alpha soon?

**Comment by Zimmi48** [comment]
Unfortunately, we are always nominating release managers too late. If only the release manager for 8.11 was already nominated, we would have someone feeling responsible for doing this. In the meantime, anyone should feel free to take the initiative and open a PR.

**Comment by ejgallego** [comment]
The checkbox for the tag is marked, however the tag is not there.

I think the right command to do is
```
git tag -s 1c8fd0f7134bcc295e31613c981ef4ef2c21af35
```
but please confirm.

**Comment by Zimmi48** [comment]
The checkbox is marked because this issue is about 8.10 but here we are talking about the 8.11+alpha tag.

**Comment by Zimmi48** [comment]
I don't have the same commit than you for the start of the 8.11. Like the previous time, I tend to view the first commit on this branch as the merge commit, in this case 1c8fd0f7134bcc295e31613c981ef4ef2c21af35.

**Comment by Zimmi48** [comment]
Don't forget to always use `git push --dry --tags` before running `git push --tags`.

**Comment by ejgallego** [comment]
@Zimmi48 isn't the commit you have written down the same I proposed?

**Comment by Zimmi48** [comment]
Sorry, my copy-pasting skills have failed me. I meant: be6f3a6234ee809dd3c290621d80c3280a41355e

**Comment by ejgallego** [comment]
Thanks @Zimmi48 , indeed I had this doubt, but however it seems strange to me that the choice is the merge commit, in the sense that for example if we bisect, we can be for example in 1c8fd0f7134bcc295e31613c981ef4ef2c21af35 and that's already 8.11 territory, however `git describe` will not work properly, right?

**Comment by ejgallego** [comment]
So indeed:
```
$ git tag -s V8.11+alpha be6f3a6234ee809dd3c290621d80c3280a41355e
$ git describe
  V8.11+alpha-69-g11dd3aeedd

$ git checkout 1c8fd0f7134bcc295e31613c981ef4ef2c21af35
$ git describe
  V8.10+alpha-2370-g1c8fd0f713
```
IMVHO this is buggy.

**Comment by Zimmi48** [comment]
Good point.

**Comment by ejgallego** [comment]
> The checkbox is marked because this issue is about 8.10 but here we are talking about the 8.11+alpha tag.

Well, the checkbox does indeed refer to the v8.11 tag, as the tag has to happen after branching. So I am not sure I understand what you mean.

> Good point.

Yup, indeed I think in this case the "the first commit to be in `master` and not in the branch` is indeed the one I pointed out, not the merge commit. I'd wait a bit to tag as to gather opinions from @maximedenes which used to know this stuff well.

**Comment by Zimmi48** [comment]
> Well, the checkbox does indeed refer to the v8.11 tag, as the tag has to happen after branching. So I am not sure I understand what you mean.

It might not be fully clear, but given that I've written this doc, I'm pretty sure of what I meant. This is further emphasized by the fact that this is at the beginning of the checklist and that the title says "previous version".

**Comment by Zimmi48** [comment]
So in this particular case "previous version" = 8.9.

**Comment by ejgallego** [comment]
Oh indeed @Zimmi48 I am confused for the reason that most checkboxes were unchecked when they should be; also because the Coq 8.11 issue has not been created, right?

**Comment by Zimmi48** [comment]
Right, in principle this was also supposed to be the job of the 8.11 RM...

**Comment by ejgallego** [comment]
> Good point.

I guess then that we should push the tag with commit 1c8fd0f7134bcc295e31613c981ef4ef2c21af35 

Are you OK with that @Zimmi48 ? If so I'll push.

**Comment by Zimmi48** [comment]
I am.

**Comment by erikmd** [comment]
@ejgallego said
> I am confused for the reason that most checkboxes were unchecked when they should be

and so was I :) but I guess it's time to prepare a Docker image for Coq 8.10+beta1.

Concretely: the build of `coqorg/coq:8.9.1` and `coqorg/coq:8.10-beta1` is in progress, but I think that `coqorg/coq:8.10-beta1` will fail as there is no corresponding `coq-bignums` version:

```
$ opam install coq.8.10+beta1 coq-bignums
The following dependencies couldn't be met:
  - coq-bignums -> coq (< 8.10~ | >= dev) -> ocaml < 4.05.0
      base of this switch (use `--unlock-base' to force)
Your request can't be satisfied:
  - No available version of coq satisfies the constraints

No solution found, exiting
```

Fortunately there is a tag https://github.com/coq/bignums/tree/V8.10+beta1,
so I will open the related PR for coq-bignums in released.

**Comment by ejgallego** [comment]
Also, the 8.11 issue is missing and for some reason there was no call that I can remember for a 8.11 RM.

Maybe an issue should be opened about it, @Zimmi48 ?

**Comment by Blaisorblade** [comment]
8.10 was apparently released, but not on opam.

BTW, I guess the release checklist above was abandoned, but it omits releasing on opam.

**Comment by Zimmi48** [comment]
Hum, related to this, we now have a problem with this page: https://coq.inria.fr/opam-using.html
The mentioned version of OCaml is not supported anymore, *and* the 8.10.0 package is not available (yet).

**Comment by Blaisorblade** [comment]
Getting the opam package in doesn't seem trivial, it seems blocked on https://github.com/ocaml/opam-repository/issues/15027, tho @clarus is helping.

Those eager to use opam can use

```
opam repo add coq-released-fork https://github.com/Blaisorblade/opam-repository.git#coq-8.10
coq install coq.8.10.0
coq install coqide.8.10.0 # If wanted
```

That passed CI on most distros, but no guarantees. And that branch will likely be deleted when the PR is merged.

**Comment by Zimmi48** [comment]
You might want to advertise this on the Coq-Club thread, so that people are aware.
